### PR TITLE
[Bug Fix] Fix #door Save

### PR DESCRIPTION
**Pull request by Kinglykrab**
# Notes
- Using `#door create` then `#door save` was overwriting doors instead of using the next highest ID.
- Remove the following unused methods.
```cpp
uint32 GetGuildEQID(uint32 guilddbid);
void UpdateDoorGuildID(int doorid, int guild_id);
int32 GetDoorsCount(uint32* oMaxID, const char *zone_name, int16 version);```

### CODE REVIEW

File: zone/doors.cpp [hunk 1]
```diff
@@ -685,82 +687,24 @@ void Doors::ToggleState(Mob *sender)
...
-}
-
-int32 ZoneDatabase::GetDoorsDBCountPlusOne(const char *zone_name, int16 version)
-{
-
-	uint32 oMaxID = 0;
-
-	std::string query   = StringFormat(
+	const auto query = fmt::format(
 		"SELECT MAX(doorid) FROM doors "
```
Comment (Akkadius) [inline zone/doors.cpp:700]: Max can be null, shouldn't assume `std::stoi` will convert safely on its own in this edge case

File: zone/doors.cpp [hunk 2]
```diff
@@ -685,82 +687,24 @@ void Doors::ToggleState(Mob *sender)
...
-	auto        results = QueryDatabase(query);
-	if (!results.Success()) {
-		return -1;
-	}
-
-	if (results.RowCount() != 1) {
-		return -1;
+	auto results = QueryDatabase(query);
+	if (!results.Success() || !results.RowCount()) {
+		return 255;
```
Comment (Akkadius) [inline zone/doors.cpp]: 255 ?

Comment (Kinglykrab) [reply zone/doors.cpp]: Changed to `-1` like it was previously.

File: zone/doors.cpp (hunk 1 above)
Comment (Kinglykrab) [reply zone/doors.cpp:700]: I'll use COALESCE here similar to the other methods I've seen.

File: zone/doors.cpp [hunk 3]
```diff
@@ -685,81 +687,23 @@ void Doors::ToggleState(Mob *sender)
...
 }
 
-int32 ZoneDatabase::GetDoorsCountPlusOne(const char *zone_name, int16 version)
+int ZoneDatabase::GetDoorsDBCountPlusOne(std::string zone_short_name, int16 version)
 {
-	std::string query   = StringFormat(
-		"SELECT MAX(id) FROM doors WHERE zone = '%s' AND version = %u",
-		zone_name,
+	const auto query = fmt::format(
+		"SELECT COALESCE(MAX(doorid), 0) FROM doors "
```
Comment (Akkadius) [inline zone/doors.cpp:700]: Wouldn't you want this value to be at least `1` instead of `0`

**Review by Akkadius — APPROVED** [review]
Minor comment, lgtm

File: zone/doors.cpp (hunk 3 above)
Comment (Kinglykrab) [reply zone/doors.cpp:700]: True enough.

**Pull request merged**